package/README.adoc

```diff
@@ -0,0 +1,140 @@
+                                                   |
+                                                   |
+                                                   |
+                                                   |
+                       +---------------------------+--------------------------+
+                       |                           |                          |
+                       |                           |                          |
+                       |                           |                          |
+              +--------v---------+       +---------v--------+       +---------v--------+
+              |                  |       |                  |       |                  |
+              |                  |       |                  |       |                  |
+              | Deployed Server  |       | Deployed Server  |       | Deployed Server  |
+              |                  |       |                  |       |                  |
+              |                  |       |                  |       |                  |
+              +------------------+       +------------------+       +------------------+
+
+
+`thing-flinger` defines three types of nodes:
+ * Client - Where a user typically edits their code and runs thing-flinger. This can run any OS.
+ * Builder - A Helios box where Omicron is built and packaged
+ * Deployed Server - Helios machines where Omicron will be installed and run
+
+It's not at all necessary for these to be separate nodes. For example, a client and builder can be
+the same machine, as long as it's a Helios box. Same goes for Builder and a deployment server. The
+benefit of this separation though, is that it allows editing on something like a laptop, without
+having to worry about setting up a development environment on an illumos based host.
+
+Machine topology is configured in a `TOML` file that is passed on the command line. All illumos
+machines are listed under `servers`, and just the names are used to for configuring a builder and
+deployment servers. An link:src/bin/deployment-example.toml[example] is provided.
+
+Thing flinger works over SSH, and so the user must have the public key of their client configured
+for their account on all servers. SSH agent forwarding is used to prevent the need for the keys of
+the builder to also be on the other servers, thus minimizing needed server configuration.
+
+== Typical Workflow
+
+=== Prerequisites
+
+Ensure you have an account on all illumos boxes, with the client public key in
+`~/.ssh/authorized_keys`.
+
+.The build machine must have Rust and cargo installed, as well as
+all the dependencies for Omicron installed. Following the *prerequisites* in the
+https://github.com/oxidecomputer/omicron/#build-and-run[Build and run] section of the main Omicron
+README is probably a good idea.
+
+=== Command Based Workflow
+
+==== Build omicron-package on client. 
+
+`cargo build -p omicron-package`
```

Comment:
Should this be instructions for building `thing-flinger`? That's what all the following commands seem to be using.